rlmolecule/alphazero/alphazero.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+import numpy as np
+
+from rlmolecule.alphazero.alphazero_problem import AlphaZeroProblem
+from rlmolecule.alphazero.alphazero_vertex import AlphaZeroVertex
+from rlmolecule.mcts.mcts import MCTS
+from rlmolecule.tree_search.graph_search_state import GraphSearchState
+
+logger = logging.getLogger(__name__)
+
+
+class AlphaZero(MCTS):
+    """
+    This class defines the interface for implementing AlphaZero-based games within this framework.
+    Such a game overrides the abstract methods below with application-specific implementations.
+    
+    AlphaZeroGame interacts with AlphaZeroVertex. See AlphaZeroVertex for more details.
+    """
+
+    def __init__(self,
+                 problem: AlphaZeroProblem,
+                 num_mcts_samples: int = 20,
+                 min_reward: float = 0.0,
+                 pb_c_base: float = 1.0,
+                 pb_c_init: float = 1.25,
+                 dirichlet_noise: bool = True,
+                 dirichlet_alpha: float = 1.0,
+                 dirichlet_x: float = 0.25,
+                 ) -> None:
+        """
+        Constructor.
+        :param min_reward: Minimum reward to return for invalid actions
+        :param pb_c_base: 19652 in pseudocode
+        :param pb_c_init:
+        :param dirichlet_noise: whether to add dirichlet noise
+        :param dirichlet_alpha: dirichlet 'shape' parameter. Larger values spread out probability over more moves.
+        :param dirichlet_x: percentage to favor dirichlet noise vs. prior estimation. Smaller means less noise
+        """
+        super().__init__(problem, num_mcts_samples=num_mcts_samples)
+        self._min_reward: float = min_reward
+        self._pb_c_base: float = pb_c_base
+        self._pb_c_init: float = pb_c_init
+        self._dirichlet_noise: bool = dirichlet_noise
+        self._dirichlet_alpha: float = dirichlet_alpha
+        self._dirichlet_x: float = dirichlet_x
+
+    @property
+    def problem(self) -> AlphaZeroProblem:
+        # noinspection PyTypeChecker
+        return self._problem
```

Comment:
Is this only needed for type checking?

Reply:
Yeah, I think the way you have it is fine, I'm still coming up to speed on typing so I was just double-checking.